lib/features/coin_flow/cubit/search_coin_cubit.dart

```diff
@@ -4,13 +4,27 @@ import 'package:equatable/equatable.dart';
 part 'search_coin_state.dart';
 
 class SearchCoinCubit extends Cubit<SearchCoinState> {
-  SearchCoinCubit() : super(const SearchCoinInitialState());
+  SearchCoinCubit()
+      : super(SearchCoinState(
+            status: CoinAnimationStatus.initial, stopwatch: Stopwatch()));
 
   static const searchDuration = Duration(milliseconds: 2000);
 
-  Future<void> searchForCoin() async {
-    emit(const SearchCoinAnimationState());
-    await Future.delayed(searchDuration);
-    emit(const SearchCoinFoundedState());
+  void startAnimation() async {
+    emit(state.copyWith(
+      status: CoinAnimationStatus.initial,
+    ));
+    await Future.delayed(const Duration(milliseconds: 50));
```

Comment:
I know this line is strange, but its the only way i could manage to trigger the SearchCoinCubit listener on search_for_coin_screen.dart. If you have another fix or an idea as to why please let me know.

I think it has to do with triggering this function in the router while the page builds. [🤔](https://emojipedia.org/thinking-face)